### ISSUE TITLE
[PATCH] Silence diag()s in ext/XS-APItest/t/blockhooks.t during 'make  test'

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10485]
Migrated from [rt.perl.org#76480](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=76480) (status was 'resolved')

Searchable as RT76480$

**Comment by p5pRT** [comment]
### From @jdhedden

#### Created by @jdhedden



ext/XS\-APItest/t/blockhooks\.t has some diag\(\)s that pollute the
output of 'make test'&#8203;:

ext/XS\-APItest/t/blockhooks\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\# \#\#
COMPILE TIME \#\#
\# \#\# RUN TIME \#\#
ok

The attached patch cleans this up when PERL\_CORE is defined\.

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low
    module=XS::APItest

Site configuration information for perl 5.13.2:

Configured by Jerry at Mon Jul 12 11:40:52 EDT 2010.

Summary of my perl5 (revision 5 version 13 subversion 2 patch 41835)
configuration:
  Snapshot of: 9df83ffded21f5c1d01d1ef847c3447b6553810f
  Platform:
    osname=cygwin, osvers=1.5.25(0.15642), archname=cygwin-thread-multi-64int
    uname='cygwin_nt-5.1 pn100-02-2-364p 1.5.25(0.15642) 2008-06-12
19:34 i686 cygwin '
    config_args='-de -Dusedevel -Dversiononly=no -Dinstallusrbinperl
-Uusemallocwrap -Duse64bitint -Dusethreads -Dinc_version_list=none
-Dnoextensions=IPC/SysV Sys/Syslog Devel/DProf Devel/Peek re GDBM_File
NDBM_File ODBM_File Text/Soundex Math/BigInt/FastCalc Time/Piece -A
append:ccflags= -DNO_MATHOMS  -A define:optimize=-Os -pipe
-funit-at-a-time -march=pentium4 -mfpmath=sse -mieee-fp -mmmx -msse
-msse2'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=undef, uselongdouble=undef
    usemymalloc=y, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-DPERL_USE_SAFE_PUTENV -U__STRICT_ANSI__
-DNO_MATHOMS -fno-strict-aliasing -pipe',
    optimize='-Os -pipe -funit-at-a-time -march=pentium4 -mfpmath=sse
-mieee-fp -mmmx -msse -msse2',
    cppflags='-DPERL_USE_SAFE_PUTENV -U__STRICT_ANSI__ -DNO_MATHOMS
-fno-strict-aliasing -pipe'
    ccversion='', gccversion='3.4.4 (cygming special, gdc 0.12, using
dmd 0.125)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long long', ivsize=8, nvtype='double', nvsize=8,
Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='g++', ldflags =' -Wl,--enable-auto-import
-Wl,--export-all-symbols -Wl,--stack,8388608
-Wl,--enable-auto-image-base -s -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib /lib
    libs=-lgdbm -ldl -lcrypt -lgdbm_compat
    perllibs=-ldl -lcrypt
    libc=/usr/lib/libc.a, so=dll, useshrplib=true, libperl=cygperl5_13_2.dll
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' -s'
    cccdlflags=' ', lddlflags=' --shared  -Wl,--enable-auto-import
-Wl,--export-all-symbols -Wl,--stack,8388608
-Wl,--enable-auto-image-base -s -L/usr/local/lib'

Locally applied patches:



@INC for perl 5.13.2:
    /usr/lib/perl5/site_perl/5.13.2/cygwin
    /usr/lib/perl5/site_perl/5.13.2
    /usr/lib/perl5/5.13.2/cygwin
    /usr/lib/perl5/5.13.2
    .


Environment for perl 5.13.2:
    CYGWIN=ntsec
    HOME=/home/jhedden
    LANG=C
    LANGUAGE=C
    LC_ALL=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/jhedden/bin:/usr/local/src/perl/bin:/link/work/bin:/usr/local/bin:/usr/bin:/bin:/usr/X11R6/bin:/c/Program
Files/WiX:/c/djgpp/bin:/c/bb/tools/apache-ant-1.6.5/bin:/c/Program
Files/nant-0.85/bin:/c/bb/tools/jse5/bin:/c/dev-cpp/bin:/c/WINDOWS/system32:/c/WINDOWS:/c/WINDOWS/system32/WBEM:/c/blp/API/dde:/c/blp/API:/c/oracle/ora92/bin:/c/Program
Files/Oracle/jre/1.3.1/bin:/c/Program
Files/Oracle/jre/1.1.8/bin:/c/Program
Files/Hummingbird/Connectivity/7.10/Accessories:/usr/bin:/c/Program
Files/Windows Imaging/:/c/Program Files/Citrix/ICAService/:/c/Program
Files/Citrix/System32/:.
    PERLIO=perlio
    PERL_BADLANG (unset)
    SHELL (unset)

```
</details>

**Comment by p5pRT** [comment]
### From @jdhedden

<details><summary>0001-Silence-diag-s-in-ext-XS-APItest-t-blockhooks.t-dur.patch</summary>

```diff
From 45c6bdd51c8dd23981e97dd64e025d0d09e7fe34 Mon Sep 17 00:00:00 2001
From: Jerry D. Hedden <jdhedden@cpan.org>
Date: Mon, 12 Jul 2010 15:46:10 -0400
Subject: [PATCH] Silence diag()s in ext/XS-APItest/t/blockhooks.t during 'make test'

---
 ext/XS-APItest/t/blockhooks.t |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ext/XS-APItest/t/blockhooks.t b/ext/XS-APItest/t/blockhooks.t
index a39c3f5..c46c578 100644
--- a/ext/XS-APItest/t/blockhooks.t
+++ b/ext/XS-APItest/t/blockhooks.t
@@ -13,8 +13,8 @@ BEGIN { package XS::APItest; *main::bhkav = \@XS::APItest::bhkav }
 # 'no t::BHK' switches recording off again.
 # 'use t::BHK push => "foo"' pushes onto @bhkav
 
-BEGIN { diag "## COMPILE TIME ##" }
-diag "## RUN TIME ##";
+BEGIN { diag("## COMPILE TIME ##") if (! $ENV{PERL_CORE}); }
+diag("## RUN TIME ##") if (! $ENV{PERL_CORE});
 
 use t::BHK;
     1;
-- 
1.6.1.2


```
</details>

**Comment by p5pRT** [comment]
### From ben@morrow.me.uk

Quoth perl5\-porters@&#8203;perl\.org&#8203;:
> 
> ext/XS\-APItest/t/blockhooks\.t has some diag\(\)s that pollute the
> output of 'make test'&#8203;:
> 
> ext/XS\-APItest/t/blockhooks\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\# \#\#
> COMPILE TIME \#\#
> \# \#\# RUN TIME \#\#
> ok
> 
> The attached patch cleans this up when PERL\_CORE is defined\.

Sorry\, that was me\. There's no need to check for PERL\_CORE\, XS&#8203;::APItest
is never on CPAN\.

Ben

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @iabyn

fixed with 72a21adedd766a965f0f4129023b8e664137f298

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
